dcm4che-core/src/main/java/org/dcm4che3/io/DicomInputStream.java

```diff
@@ -678,12 +675,23 @@ public void readAttributes(Attributes attrs, int len, Predicate<DicomInputStream
                 excludeBulkData = includeBulkData == IncludeBulkData.NO && isBulkData(attrs);
                 includeBulkDataURI = length != 0 && vr != VR.SQ
                         && includeBulkData == IncludeBulkData.URI && isBulkData(attrs);
+                if (length < -1 && !excludeBulkData && !includeBulkDataURI) {
+                    throw tagValueTooLargeException();
+                }
                 handler.readValue(this, attrs);
             } else
                 skipAttribute(UNEXPECTED_ATTRIBUTE, "readAttributes()");
         }
     }
 
+    private static long toLongLength(int len) {
+        return len & 0xffffffffL;
```

Comment:
We should consider the special -1 case here for the conversion:
`if( length == -1 ) return -1;`